File: tools/bench/answers/go_haiku/s3/09_topwords.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

func main() {
	sentence := "the cat sat on the mat with the cat"
	words := strings.Fields(sentence)
	freq := make(map[string]int)
	for _, word := range words {
		freq[word]++
	}
	type kv struct {
		word  string
		count int
	}
	var pairs []kv
	for word, count := range freq {
		pairs = append(pairs, kv{word, count})
	}
	sort.Slice(pairs, func(i, j int) bool {
		return pairs[i].count > pairs[j].count
	})
	for i := 0; i < 2 && i < len(pairs); i++ {
		fmt.Printf("%s %d\n", pairs[i].word, pairs[i].count)
	}
}

File: tools/bench/answers/python_haiku/s2/09_topwords.py
s = "the cat sat on the mat with the cat"
words = s.split()
freq = {}
for word in words:
    freq[word] = freq.get(word, 0) + 1
sorted_words = sorted(freq.items(), key=lambda x: -x[1])
for i in range(2):
    print(f"{sorted_words[i][0]} {sorted_words[i][1]}")
